internal/sniproxy/sniproxy.go

```diff
@@ -264,11 +266,20 @@ func (p *SNIProxy) shouldForward(ctx *SNIContext) (ok bool) {
 	return false
 }
 
+type writeCloser interface {
+	CloseWrite() error
+}
+
 // copy copies data from src to dst and signals that the work is done via the
 // wg wait group.
 func tunnel(ctx *SNIContext, dst net.Conn, src io.Reader) (written int64) {
 	defer func() {
-		_ = dst.(*net.TCPConn).CloseWrite()
+		switch c := dst.(type) {
```

Comment:
This change is required because with HTTPS proxies one side of connection is actually `*tls.Conn`.